astroid/tests/unittest_brain.py

```diff
@@ -812,5 +812,21 @@ def test_uuid_has_int_member(self):
         self.assertIsInstance(inferred, nodes.Const)
 
 
+@test_utils.require_version('3.6')
+class BrainTypingTest(unittest.TestCase):
+    def test_typing_types(self):
+        ast_nodes = builder.extract_node("""
+        from typing import TypeVar, Iterable, Tuple, NewType, Dict, Union
+        TypeVar('MyTypeVar', int, float, complex) #@
+        Iterable[Tuple[MyTypeVar, MyTypeVar]] #@
+        TypeVar('AnyStr', str, bytes) #@
+        NewType('UserId', str) #@
```

Comment:
NewType seems to be a function rather than a ClassDef, I'm confused to why we have to infer it as a class. And nonetheless, I'm not sure implementing it in this way brings up a lot of benefits, we also have to make it act as a subclass of the original types that were passed in the constructor, which we don't do currently.